modules/dag-l1/src/test/scala/org/tessellation/dag/l1/domain/snapshot/programs/SnapshotProcessorSuite.scala

```diff
@@ -73,84 +73,92 @@ object SnapshotProcessorSuite extends SimpleIOSuite with TransactionGenerator {
   def testResources: Resource[IO, TestResources] =
     SecurityProvider.forAsync[IO].flatMap { implicit sp =>
       KryoSerializer.forAsync[IO](Main.kryoRegistrar ++ nodeSharedKryoRegistrar).flatMap { implicit kp =>
-        Random.scalaUtilRandom[IO].asResource.flatMap { implicit random =>
-          for {
-            balancesR <- Ref.of[IO, Map[Address, Balance]](Map.empty).asResource
-            blocksR <- MapRef.ofConcurrentHashMap[IO, ProofsHash, StoredBlock]().asResource
-            lastSnapR <- SignallingRef.of[IO, Option[(Hashed[GlobalIncrementalSnapshot], GlobalSnapshotInfo)]](None).asResource
-            lastAccTxR <- MapRef.ofConcurrentHashMap[IO, Address, LastTransactionReferenceState]().asResource
-            waitingTxsR <- MapRef.ofConcurrentHashMap[IO, Address, NonEmptySet[Hashed[Transaction]]]().asResource
-            transactionStorage = new TransactionStorage[IO](lastAccTxR, waitingTxsR, TransactionReference.empty)
-            validators = SharedValidators.make[IO](None, None, Some(Map.empty), Long.MaxValue)
-
-            currencySnapshotAcceptanceManager = CurrencySnapshotAcceptanceManager.make(
-              BlockAcceptanceManager.make[IO](validators.currencyBlockValidator),
-              Amount(0L)
-            )
-            currencyEventsCutter = CurrencyEventsCutter.make[IO]
-            currencySnapshotCreator = CurrencySnapshotCreator
-              .make[IO](currencySnapshotAcceptanceManager, None, SnapshotSizeConfig(Long.MaxValue, Long.MaxValue), currencyEventsCutter)
-            currencySnapshotValidator = CurrencySnapshotValidator.make[IO](currencySnapshotCreator, validators.signedValidator, None, None)
-
-            currencySnapshotContextFns = CurrencySnapshotContextFunctions.make(currencySnapshotValidator)
-            globalSnapshotStateChannelManager <- GlobalSnapshotStateChannelAcceptanceManager.make[IO](None, NonNegLong(10L)).asResource
-            jsonBrotliBinarySerializer <- JsonBrotliBinarySerializer.make[IO]().asResource
-            globalSnapshotAcceptanceManager = GlobalSnapshotAcceptanceManager.make(
-              BlockAcceptanceManager.make[IO](validators.blockValidator),
-              GlobalSnapshotStateChannelEventsProcessor
-                .make[IO](
-                  validators.stateChannelValidator,
-                  globalSnapshotStateChannelManager,
-                  currencySnapshotContextFns,
-                  jsonBrotliBinarySerializer
-                ),
-              Amount(0L)
-            )
-            globalSnapshotContextFns = GlobalSnapshotContextFunctions.make[IO](globalSnapshotAcceptanceManager)
-            snapshotProcessor = {
-              val addressStorage = new AddressStorage[IO] {
-                def getState: IO[Map[Address, Balance]] =
-                  balancesR.get
+        JsonHashSerializer.forSync[IO].asResource.flatMap { implicit jhs =>
```

Comment:
Why not use the `implicit0` notation here?

Reply:
well, I can, but the kryo was already created like that, so just added another one

Reply:
In other places you did a refactor I think. Up to you.